server/src/main/java/org/opensearch/index/engine/InternalEngine.java

```diff
@@ -716,11 +717,13 @@ private ExternalReaderManager createReaderManager(RefreshWarmerListener external
     }
 
     private DirectoryReader getDirectoryReader() throws IOException {
-        // for segment replication: replicas should create the reader from store, we don't want an open IW on replicas.
+        // for segment replication: replicas should create the reader from store and, we don't want an open IW on replicas.
+        /* We should always wrap replicas with a SoftDeletesDirectoryReaderWrapper as we use soft deletes when segment replication is on for
```

Comment:
nit - "We should always wrap the DirectoryReader used on replicas with `SoftDeletesDirectoryReaderWrapper` so that we filter out soft deletes.